docs/develop/environment-configuration.mdx

```diff
@@ -330,6 +330,61 @@ public static class LoadFromFile
 ```
 
   </SdkTabs.DotNet>
+
+  <SdkTabs.TypeScript>
+To load the `default` profile along with any environment variables in TypeScript, use the `loadClientConnectConfig` helper from `@temporalio/envconfig` package.
+
+{/* SNIPSTART typescript-env-config-load-profile-with-overrides {"highlightedLines": "15-18,30-31"} */}
+[env-config/src/load-profile.ts](https://github.com/temporalio/samples-typescript/blob/main/env-config/src/load-profile.ts)
+
+```ts {15-18,30-31}
+import { Connection, Client } from '@temporalio/client';
+import { loadClientConnectConfig } from '@temporalio/envconfig';
+import { resolve } from 'path';
+
+async function main() {
+  console.log("--- Loading 'staging' profile with programmatic overrides ---");
+
+  const configFile = resolve(__dirname, '../config.toml');
+  const profileName = 'staging';
+
+  // The 'staging' profile in config.toml has an incorrect address (localhost:9999)
+  // We'll programmatically override it to the correct address
+
+  // Load the 'staging' profile.
+  const config = loadClientConnectConfig({
+    profile: profileName,
+    configSource: { path: configFile },
+  });
+
+  // Override the target host to the correct address.
+  // This is the recommended way to override configuration values.
+  config.connectionOptions.address = 'localhost:7233';
+
+  console.log(`\nLoaded '${profileName}' profile from ${configFile} with overrides.`);
+  console.log(`  Address: ${config.connectionOptions.address} (overridden from localhost:9999)`);
```

Comment:
I don't think we need to do this override bit, I would follow the simpler sample here, imo.